avro-serde/src/main/java/io/confluent/kafka/streams/serdes/GenericAvroSerde.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.util.Collections;
+import java.util.Map;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+
+/**
+ * A serde (serializer/deserializer) for Apache Kafka's Streams API that can be used for for reading
+ * and/or writing data in "generic Avro" format.
+ *
+ * <p>Usage example for configuring this serde as a Kafka Streams application's default serde for
+ * both record keys and record values:
+ *
+ * <p><pre>{@code
+ * Properties streamsConfiguration = new Properties();
+ * streamsConfiguration.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, GenericAvroSerde.class);
+ * streamsConfiguration.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, GenericAvroSerde.class);
+ * streamsConfiguration.put(
+ *     AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+ *     "http://confluent-schema-registry-server:8081/");
+ * }</pre>
+ *
+ * <p>Usage example for explicitly overriding the application's default serdes (whatever they were
+ * configured to) so that only specific operations such as {@code KStream#to()} use this serde:
+ *
+ * <p><pre>{@code
+ * Serde<GenericRecord> genericAvroSerde = new GenericAvroSerde();
+ * boolean isKeySerde = false;
+ * genericAvroSerde.configure(
+ *     Collections.singletonMap(
+ *         AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+ *         "http://confluent-schema-registry-server:8081/"),
+ *     isKeySerde);
+ * KStream<String, GenericRecord> stream = ...;
+ * stream.to(Serdes.String(), genericAvroSerde, "my-output-topic");
+ * }</pre>
+ */
+public class GenericAvroSerde implements Serde<GenericRecord> {
+
+  private final Serde<GenericRecord> inner;
+
+  /**
+   * Implementation detail: This constructor is used by Kafka's Streams API.
+   */
+  public GenericAvroSerde() {
+    inner = Serdes.serdeFrom(new GenericAvroSerializer(), new GenericAvroDeserializer());
+  }
+
+  public GenericAvroSerde(final SchemaRegistryClient client) {
+    this(client, Collections.<String, Object>emptyMap(), Collections.<String, Object>emptyMap());
+  }
+
+  public GenericAvroSerde(final SchemaRegistryClient client,
+                          final Map<String, ?> serializerConfig,
+                          final Map<String, ?> deserializerConfig) {
+    if (client == null) {
+      throw new IllegalArgumentException("schema registry client must not be null");
+    }
+    inner = Serdes.serdeFrom(
+        new GenericAvroSerializer(client, serializerConfig),
+        new GenericAvroDeserializer(client, deserializerConfig));
+  }
+
+  @Override
+  public Serializer<GenericRecord> serializer() {
+    return inner.serializer();
+  }
+
+  @Override
+  public Deserializer<GenericRecord> deserializer() {
+    return inner.deserializer();
+  }
+
+  @Override
+  public void configure(final Map<String, ?> serdeConfig, final boolean isSerdeForRecordKeys) {
+    inner.serializer().configure(serdeConfig, isSerdeForRecordKeys);
```

Comment:
The various `configure()` methods aren't getting test coverage due to using the shorthand constructors. I'd suggest flipping that even if we don't remove those constructors since using `configure()` is the path that is used under normal circumstances.